### PR TITLE
Account for nullability in array conversion to interface

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/34827

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -95147,5 +95147,59 @@ public override void M<U>(B x, U y)
...
+        [Fact, WorkItem(34827, "https://github.com/dotnet/roslyn/issues/34827")]
+        public void Verify34827_Nested()
+        {
+            string source = @"
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        IEnumerable<object[]> x1 = new[] { new[] { ""string"" }, null }; // 1
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >new[] { new[] { ""string"" }, null } [](start = 35, length = 36)

It doesn't look like this scenario is targeting nesting case, or I misunderstood what "Nested" stands for in the name of the test.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 2]
```diff
@@ -95147,5 +95147,59 @@ public override void M<U>(B x, U y)
             // Per https://github.com/dotnet/roslyn/issues/35075 errors should be expected
             c.VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(34827, "https://github.com/dotnet/roslyn/issues/34827")]
+        public void Verify34827()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >Verify34827 [](start = 20, length = 11)

Consider giving these tests descriptive names. Perhaps "ArrayConversionToInterface...".

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 3]
```diff
@@ -95147,5 +95147,59 @@ public override void M<U>(B x, U y)
...
+        {
+            string source = @"
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        IEnumerable<object> x1 = new[] { ""string"", null }; // 1
+        IEnumerable<object?> x2 = new[] { ""string"", null };
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >, null [](start = 52, length = 6)

Consider dropping `null` from the cases where the target has a nullable element type.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: Fixed. Thanks

---
In reply to: [281739109](https://github.com/dotnet/roslyn/pull/35541#discussion_r281739109) [](ancestors = 281739109)

**Comment by jcouv** [comment]
Rebased to resolve conflicts

**Pull request merged**